subprojects/scala/src/main/java/org/gradle/language/scala/tasks/AbstractScalaCompile.java

```diff
@@ -252,6 +253,7 @@ private void configureIncrementalCompilation(ScalaCompileSpec spec) {
      * {@inheritDoc}
      */
     @Override
+    @IgnoreEmptyDirectories
     @PathSensitive(PathSensitivity.NAME_ONLY)
     public FileTree getSource() {
```

Comment:
`AbstractScalaCompile` is a source task, so it already has `@IgnoreEmptyDirectories`.